packages/browser/src/plugins/ajs-destination/index.ts

```diff
@@ -139,22 +140,20 @@ export class LegacyDestination implements DestinationPlugin {
       analyticsInstance
     )
 
-    this.onReady = new Promise((resolve) => {
-      const onReadyFn = (): void => {
-        this._ready = true
-        resolve(true)
-      }
+    setTimeout(() => {
+      const e = 'Destination timed out'
+      this.initializePromise.reject(e)
+      this.readyPromise.reject(e)
+    }, this.options.destinationTimeout!)
 
-      this.integration!.once('ready', onReadyFn)
+    this.integration!.once('ready', () => {
+      this._ready = true
+      this.readyPromise.resolve()
```

Comment:
to make the promises more sound maybe we should chain the state of `_ready` and `_initialized` to them during construction

```ts
constructor(...) {
	//...
	this.readyPromise.then(() => { this._ready = true }).catch(() => { this._ready = false });
	// Same for initializePromise
}
```

That way you only have to care about resolving or rejecting promises throughout the code.

Reply:
@oscb  Doesn't the meaning of .ready = true just mean ".ready has been fired", which should happens on either resolve _or_ reject? 

I feel like making .ready a tristate variable is a bit confusing (undefined, true, or false)?

Reply:
Please see the latest push for revision to this